android-components/components/browser/icons/src/main/java/mozilla/components/browser/icons/decoder/SvgIconDecoder.kt

```diff
@@ -21,13 +22,17 @@ import mozilla.components.support.images.decoder.ImageDecoder
  */
 class SvgIconDecoder(val context: Context) : ImageDecoder {
 
-    override fun decode(data: ByteArray, desiredSize: DesiredSize): Bitmap? {
-        val request = ImageRequest.Builder(context)
-            .data(data)
-            .build()
+    override fun decode(data: ByteArray, desiredSize: DesiredSize): Bitmap? =
+        try {
+            val request = ImageRequest.Builder(context)
+                .size(desiredSize.targetSize)
+                .data(data)
+                .build()
 
-        return SvgImageLoader.getInstance(context).executeBlocking(request).drawable?.toBitmap()
-    }
+            SvgImageLoader.getInstance(context).executeBlocking(request).drawable?.toBitmap()
+        } catch (e: OutOfMemoryError) {
+            null
```

Comment:
Could we also include an error log of the exception here? I see the other decoders don't but given the issues we've seen using this it may prove beneficial